pkg/app/piped/eventwatcher/eventwatcher.go

```diff
@@ -277,26 +279,45 @@ func modifyYAML(path, field, newValue string) ([]byte, bool, error) {
 	if err != nil {
 		return nil, false, fmt.Errorf("failed to replace value at %s with %s: %w", field, newValue, err)
 	}
+
 	return newYml, false, nil
 }
 
-// convertStr converts a given value into a string.
-func convertStr(value interface{}) (out string, err error) {
-	switch v := value.(type) {
-	case string:
-		out = v
-	case int:
-		out = strconv.Itoa(v)
-	case int64:
-		out = strconv.FormatInt(v, 10)
-	case uint64:
-		out = strconv.FormatUint(v, 10)
-	case float64:
-		out = strconv.FormatFloat(v, 'f', -1, 64)
-	case bool:
-		out = strconv.FormatBool(v)
+// isUpToDate check if all targets is up-to-date or not
+func isUpToDate(targets []string, newVal string) bool {
+	for _, t := range targets {
+		if t != newVal {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
it depends on whether we allow sequences as a `newVal` or not. If we support sequences, we need to parse `newVal` and compare each element precisely.
In the case that we allow only scalar value as a `newVal`, will be a kind of like:

```go
func isUpToDate(targets []string, newVal string) bool {
	if len(targets) != 1 {
		return false
	}
	return targets[0] == newVal
}
```